openml/testing.py

```diff
@@ -135,17 +172,53 @@ def _cleanup_fixture(self):
         # pauses the code execution here till all tests in the 'session' is over
         yield
         # resumes from here after all collected tests are completed
+
+        # Local file deletion
         files = os.walk(directory)
         new_file_list = []
         for root, _, filenames in files:
             for filename in filenames:
                 new_file_list.append(os.path.join(root, filename))
         # filtering the files generated during this run
         new_file_list = list(set(new_file_list) - set(old_file_list))
-        print("Files to delete in local: {}".format(new_file_list))
         for file in new_file_list:
             os.remove(file)
 
+        # Test server deletion
+        openml.config.server = TestBase.test_server
+        openml.config.apikey = TestBase.apikey
+        entity_types = list(TestBase.tracker.keys())
+        # deleting 'run' first to allow other dependent entities to be deleted
+        if 'run' in entity_types:
+            index = entity_types.index('run')
+            # putting 'run' in the start of the list
+            entity_types[0], entity_types[index] = entity_types[index], entity_types[0]
+
+        # cloning file tracker to allow deletion of entries of deleted files
+        tracker = TestBase.tracker.copy()
+        # reordering to delete sub flows later
+        if 'flow' in entity_types:
+            flows = {}
+            for entity_id, entity_name in tracker['flow']:
+                flows[entity_name] = entity_id
+            # reordering flows in descending order of their flow name lengths
+            flow_deletion_order = [flows[name] for name in sorted(list(flows.keys()),
```

Comment:
How about 

```
flow_deletion_order = [entity_id for entity_id, _ in sorted(tracker['flow'], key=lambda x: len(x[1]), reverse=True)]
```
Saves a few lines and reads a bit easier (in my opinion).

Reply:
This was a lot neater. Thanks a lot!